deploy/runtime/ui/Battery Life Options.txt

```diff
@@ -0,0 +1,307 @@
+TextColour
+4
+0
+0
+0
+255
+Bold
+2
+0
+FontSize
+3
+0
+WordWrap
+2
+1
+AlignRight
+2
+0
+AlignTop
+2
+1
+RadioChoice
+
+1
+11
+Name
+5
+18
+ui_batt_life_model
+X
+3
+42
+Y
+3
+72
+Width
+3
+339
+Height
+3
+78
+Tool Tip
+5
+0
+Selection
+3
+0
+Items
+6
+3
+Cycle and calendar degradation
+Li-ion NMC/Graphite
+Li-ion LMO/LTO
+ShowCaptions
+2
+1
+Horizontal
+2
+0
+TabOrder
+3
+1
+
+1
+ui_batt_life_model
+3
+1
+Battery life model option for user interface
+0/1/2
+Battery Life
+65
+Cycle and calendar degradation|Li-ion NMC/Graphite|Li-ion LMO/LTO
+3
+1
+1
+1
+1
+0
+RadioChoice
+
+0
+2615
+on_load{'Battery Life Options'} = define() {
+	battery_type_message();
+	life_message();
+};
+
+on_change{'ui_batt_life_model'} = define(){
+	battery_type_message();
+	life_message();
+	set_batt_life_excl_var();
+};
+
+function set_batt_life_excl_var()
+{
+	value('batt_life_excl', value('ui_batt_life_model'));
+}
+
+function life_message()
+{
+	msg = '';
+	clr = 'Black';
+	options = ['Tables','Li-ion NMC/Graphite','Li-ion LMO/LTO'];
+	if ( ${batt_chem} == 0 )
+	{
+		msg = 'Only the Cycle and Calendar Degradation option is available for lead-acid and flow battery types.';
+	}
+	elseif ( ${ui_batt_life_model} == 0 )
+	{
+		msg = 'The Cycle and Calendar Degradation inputs below define battery degradation curves.';
+	}
+	else
+	{
+		msg = 'This option uses a special battery life model for the ' + options[${ui_batt_life_model}] + ' battery type that combines cycle and calendar degradation.';
+	}
+	property('lbl_batt_life_message','Caption',msg);
+	property('lbl_batt_life_message','TextColour',clr);
+}
+
+function battery_type_message()
+{
+	clr = 'Black';
+	msg = '';
+	types = ['Lead Acid: Flooded','Lead Acid: VRLA Gel','Lead Acid: VRLA AGM','Lead Acid: Custom','Lithium Ion: Lithium Manganese Oxide (LMO/Graphite)','Lithium Ion: Lithium Iron Phosphate (LFP/Graphite)','Lithium Ion: Lithium Cobalt Oxide (LCO/Graphite)','Lithium Ion: LMO/Lithium Titanate (LMO/LTO)','Lithium Ion: Nickel Manganese Cobalt Oxide (NMC/Graphite)','Lithium Ion: Nickel Cobalt Aluminum Oxide (NCA/Graphite)','Lithium Ion: Custom','Flow Battery: Vanadium','Flow Battery: Iron'];
+	if ( ${batt_chem} == 1 ) // Li-ion
+	{
+		if ( ${batt_type} == 7 ) // LMO/LTO
+		{
+			best_life = 'Li-ion LMO/LTO';
+			if ( ${ui_batt_life_model} != 2 ) { clr = 'Red'; }
+		}
+		elseif ( ${batt_type} == 8 ) // NMC/Graphite
+		{
+			best_life = 'Li-ion NMC/Graphite'; 
+			if ( ${ui_batt_life_model} != 1 ) { clr = 'Red'; }
+		}
+		else // other Li-ion types
+		{
+			best_life = 'empirical cycle and calendar degradation'; 
+			if ( ${ui_batt_life_model} != 0 || ${batt_calendar_choice} != 1 ) { clr = 'Red'; }
+		}
+	}
+	else // lead acid and flow
+	{
+		best_life = 'cycle degradation with no calendar degradation';
+		if ( ${ui_batt_life_model} != 0 || ${batt_calendar_choice} != 0 ) { clr = 'Red'; }
```

Comment:
I can see why we wouldn't want lead acid and flow batteries using the empirical model but is there a reason we are disabling custom?

![image](https://user-images.githubusercontent.com/5530592/137405044-ec87f1d5-741e-4410-9bf5-7ee29d8cfbca.png)

A search on this suggested a lead acid battery stored at less than 100% state of charge would degrade via sulfation - though a table of days and capacity wouldn't fully model that mechanism.

Reply:
This is fixed in f3e60f26c: SAM yesno message when the user chooses the empirical option for a lead-acid or flow battery.